### PR TITLE
Edit for issue # 2

### PR DESCRIPTION
**Pull request by fffarina**

### CODE REVIEW

**Comment by fffarina** [comment]
I have added to the exit method an if statement to check if upper bound for total time steps is reached.

**Comment by fffarina** [comment]
I have added an invariant to check if "time" features as key in the stoichiometry dictionary.

File: montecarlo/ssa.py [hunk 1]
```diff
@@ -75,11 +75,17 @@ def __init__(
         initial_conditions,
         propensities,
         stoichiometry,
+        total_steps=None
         max_duration=None
     ):
         """Initialize model"""
         super().__init__(**initial_conditions)
+        for i, dictionaries in enumerate(stoichiometry):
```
Comment (wfatherley) [inline montecarlo/ssa.py]: `__init__` feels like the best member to contain reserved-word logic to me, but maybe we can raise an exception instead?    

```python
for sto in stoichiometry:
    if "time" in sto:
        raise Exception("error, 'time' can't be in stoichiometry")
```

File: montecarlo/ssa.py [hunk 2]
```diff
@@ -99,6 +105,9 @@ def exit(self):
         if self.max_duration is not None:
             if self["time"][-1] >= self.max_duration:
                 return True
+        if self.total_steps is not None:
+            if len(self["time"]) == self.total_steps:
```
Comment (wfatherley) [inline montecarlo/ssa.py:108]: My initial thought was to iterate down the `total_steps` parameter to zero, but this is much more clever and contained :)

**Review by wfatherley — CHANGES_REQUESTED** [review]

**Comment by wfatherley** [comment]
closes #2, closes #3

File: montecarlo/ssa.py (hunk 1 above)
Comment (fffarina) [reply montecarlo/ssa.py]: Yes better!

File: montecarlo/ssa.py (hunk 2 above)
Comment (fffarina) [reply montecarlo/ssa.py:108]: That is how I initially thought and implemented it in my code; a while loop that checked if the length of the "time" dictionary is less than max time_steps. I saw however that you have added the end_time invariant check in the exit method and thought to do the same.

**Comment by wfatherley** [comment]
This is really cool work, thank you for your help!

**Pull request merged**